front/app/containers/EventsPage/EventsViewer/index.tsx

```diff
@@ -51,47 +61,72 @@ const StyledPagination = styled(Pagination)`
 interface Props {
   title: string;
   fallbackMessage: string;
-  events: number[];
+  eventsTime: 'past' | 'future';
   className?: string;
 }
 
 const EVENTS_PER_PAGE = 10;
 
 const EventsViewer = memo<Props>(
-  ({ title, events, className, fallbackMessage }) => {
+  ({ title, fallbackMessage, eventsTime, className }) => {
+    const [projectIds, setProjectIds] = useState<string[]>([]);
+
+    const events = useEvents(
+      projectIds,
+      eventsTime === 'future',
+      eventsTime === 'past'
+    );
+
     const [currentPage, setCurrentPage] = useState<number>(1);
-    const [visibleEvents, setVisibleEvents] = useState<number[]>([]);
+    const [visibleEvents, setVisibleEvents] = useState<IEventData[]>([]);
+    const [eventsHaveLoaded, setEventsHaveLoaded] = useState<boolean>(false);
 
     useEffect(() => {
+      if (isNilOrError(events)) {
+        setVisibleEvents([]);
+        setEventsHaveLoaded(false);
+        return;
+      }
+
       setVisibleEvents(sliceEventsToPage(events, currentPage, EVENTS_PER_PAGE));
+      setEventsHaveLoaded(true);
     }, [events, currentPage]);
 
     return (
       <div className={className}>
-        <TopBar title={title} />
-
-        {visibleEvents.length > 0 &&
-          visibleEvents.map((e, i) => (
-            <PlaceHolder key={e} first={i === 0}>
-              {e}
-            </PlaceHolder>
-          ))}
-
-        {events.length === 0 && (
-          <NoEventsContainer>
-            <NoEventsIllustration src={noEventsIllustration} />
-
-            <NoEventsText>{fallbackMessage}</NoEventsText>
-          </NoEventsContainer>
-        )}
-
-        {events.length > 10 && (
-          <StyledPagination
-            currentPage={currentPage}
-            totalPages={getNumberOfPages(events.length, EVENTS_PER_PAGE)}
-            loadPage={setCurrentPage}
-            useColorsTheme
-          />
+        <TopBar title={title} setProjectIds={setProjectIds} />
+
+        {!eventsHaveLoaded && <StyledSpinner />}
+
+        {eventsHaveLoaded && (
+          <>
+            {visibleEvents.length > 0 &&
+              visibleEvents.map((event, i) => (
+                <StyledEventCard
+                  event={event}
+                  showProjectTitle={true}
+                  last={visibleEvents.length - 1 === i}
+                  key={event.id}
+                />
+              ))}
+
+            {visibleEvents.length === 0 && (
+              <NoEventsContainer>
+                <NoEventsIllustration src={noEventsIllustration} />
+
+                <NoEventsText>{fallbackMessage}</NoEventsText>
+              </NoEventsContainer>
+            )}
+
+            {!isNilOrError(events) && events.length > 10 && (
+              <StyledPagination
+                currentPage={currentPage}
+                totalPages={getNumberOfPages(events.length, EVENTS_PER_PAGE)}
```

Comment:
totalPages should come from the API. Will you still replace it?

Reply:
It currently does not... the `useEvents` hook only returns an array of all events that match the filter criteria (`futureOnly` / `pastOnly`)

Reply:
They haven't been implemented yet, but the API supports these parameters though: https://developers.citizenlab.co/api-docs/frontweb_api/master/events/list_all_events_of_a_project.html.

You can see how it's done in e.g. useModerations (but that one is not the best example, I'm proposing some changes in a different PR). I can help out if needed

Reply:
Oh, that's about a project. We're project agnostic here on this page. Let me check what the API looks like for this epic. One sec